client/src/main/java/org/asynchttpclient/netty/ssl/DefaultSslEngineFactory.java

```diff
@@ -44,14 +42,44 @@ private SslContext getSslContext(AsyncHttpClientConfig config) throws SSLExcepti
         if (config.isAcceptAnyCertificate())
             sslContextBuilder.trustManager(InsecureTrustManagerFactory.INSTANCE);
 
-        return sslContextBuilder.build();
+        return configureSslContextBuilder(sslContextBuilder).build();
     }
 
     @Override
     public SSLEngine newSslEngine(AsyncHttpClientConfig config, String peerHost, int peerPort) {
         // FIXME should be using ctx allocator
-        SSLEngine sslEngine = sslContext.newEngine(ByteBufAllocator.DEFAULT, peerHost, peerPort);
+        SSLEngine sslEngine = getSslContext(config).newEngine(ByteBufAllocator.DEFAULT, peerHost, peerPort);
         configureSslEngine(sslEngine, config);
         return sslEngine;
     }
+
+    /**
+     * The last step of configuring the SslContextBuilder used to create an SslContext when no context is provided in
+     * the {@link AsyncHttpClientConfig}. This defaults to no-op and is intended to be overridden as needed.
+     *
+     * @param builder builder with normal configuration applied
+     * @return builder to be used to build context (can be the same object as the input)
+     */
+    protected SslContextBuilder configureSslContextBuilder(SslContextBuilder builder) {
```

Comment:
I made this return `SCB` instead of `void` to allow someone to make a new one if they *really* needed to